flow/flow.cpp

```diff
@@ -174,7 +174,10 @@ FileOutputMode setupLogging(int mpi_rank_, const std::string& deck_filename, con
 
     std::string output_dir = cmdline_output_dir;
     if (output_dir.empty()) {
-        output_dir = absolute(path(baseName).parent_path()).string();
+        if (path(fpath).has_parent_path())
+          output_dir = absolute(path(fpath).parent_path()).string();
+        else
+          output_dir = Opm::filesystem::current_path().string();
```

Comment:
Tiny nit: `fpath` already is a `path` object by construction.  You don't need to construct another `path` object here.  On another note, I think I would probably use the ternary operator:
```C++
output_dir = fpath.has_parent()
    ? absolute(fpath.parent_path()).string()
    : Opm::filesystem::current_path().string();
```
just to underscore the fact that we're assigning a value to `output_dir` in both cases.  I might also consider using `generic_string()` instead of just `.string()`.

Reply:
all your nits are belong to you.